zerver/actions/message_send.py

```diff
@@ -1603,13 +1612,22 @@ def internal_prep_private_message(
         addressee=addressee,
         content=content,
         mention_backend=mention_backend,
+        disable_external_notifications=disable_external_notifications,
     )
 
 
 def internal_send_private_message(
-    sender: UserProfile, recipient_user: UserProfile, content: str
+    sender: UserProfile,
+    recipient_user: UserProfile,
+    content: str,
+    disable_external_notifications: bool = False,
```

Comment:
We should use a "mandatory keyword argument" here -- i.e. `content: str, *, disable_external_notifications: bool = False`, and callers that want to set this parameter should pass it explicitly as `disable_external_notificaitons=True`. This pattern makes for MUCH more readable code, since you can see what this `True` does without going and reading the function it's calling. 

The same applies for all the other `internal_send_*|internal_prep_*` functions -- all the parameters after `content` should have this rule. It might make sense to do a preparatory refactor commit to make the existing parameters for the other functions that have them follow this pattern, to keep the new code simple.